peartree/toolkit.py

```diff
@@ -328,16 +323,11 @@ def coalesce(G_orig: nx.MultiDiGraph, resolution: float) -> nx.MultiDiGraph:
     # Also make sure to update the new nodes with their summary
     # stats and locational data
     for i, node in new_node_coords.items():
-        # Some nodes are completely dropped in this operation
-        # with no replacement edges (e.g. nodes that would have
-        # connected to another node that ended up getting coalesced
-        # into the same single node)
-        if i not in G.nodes():
-            continue
-
-        # For all other nodes, preserve them by re-populating
-        for key in node:
-            G.nodes[i][key] = node[key]
+        # Note: is it possible at all that node i is not in G?
```

Comment:
I dont think it is possible. but i see that you had a check on this. Could you clarify?

Reply:
Hrm I think we ran into this though I can't recall at the moment